lib/kvdb/indent.c

```diff
@@ -0,0 +1,158 @@
+#include <hse/logging/logging.h>
+#include <hse/mpool/mpool.h>
+#include <hse/pidfile/pidfile.h>
+#include <hse/rest/headers.h>
+#include <hse/rest/params.h>
+#include <hse/rest/request.h>
+#include <hse/rest/response.h>
+#include <hse/rest/server.h>
+#include <hse/rest/status.h>
+#include <hse/util/err_ctx.h>
+#include <hse/util/event_counter.h>
+#include <hse/util/mutex.h>
+#include <hse/util/platform.h>
+#include <hse/util/vlb.h>
+#include <hse/util/compiler.h>
+
+#define HSE_KVS_PUT_MASK       1
+#define HSE_KVS_PUT_VCOMP_MASK 2
+
+char *
+some_long_function_name(void *);
+
+bool long_variable_boolean;
+int long_variable_integer;
+void * long_variable_pointer;
+struct foobar_foobar;
+void * const * yoyo;
+
+
+struct foo {
+    char foobar_foobar_foobar[1234];
+    int * foo;
+    unsigned long ** bar;
+};
+
+
+// Func Decls
+int
+function1(unsigned int * arg1);
+
+int
+function2(unsigned int * arg1, unsigned int ** arg2, unsigned int *** arg3, unsigned int arg4);
+
+int
+function3(
+    unsigned int * arg1,
+    unsigned int ** arg2,
+    unsigned int *** arg3,
+    unsigned int arg4,
+    unsigned int arg5)
+{
+    *arg1 = 0;
+    **arg2 = 1;
+    return 0;
+}
+
+
+// Func Decls w/ Attributes
+
+void HSE_NORETURN
+fn_attr1(void);
+
+HSE_NORETURN void
+fn_attr2(void);
+
+void HSE_PRINTF(1, 2)
+fn_attr3(char * fmt);
+
+void HSE_NORETURN
+fn_attr4(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+HSE_NORETURN void
+fn_attr5(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+void HSE_PRINTF(1, 2)
+fn_attr6(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar * foobar);
+
+
+hse_err_t
+long_if_expr(int flags)
+{
+    if (long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL ||
+        (flags & HSE_KVS_PUT_VCOMP_MASK) == HSE_KVS_PUT_VCOMP_MASK)
+        return merr(EINVAL);
+
+    if (HSE_UNLIKELY(
+            long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL ||
+            (flags & HSE_KVS_PUT_VCOMP_MASK) == HSE_KVS_PUT_VCOMP_MASK))
+        return merr(EINVAL);
+
+    if (long_variable_boolean || long_variable_integer > 0 ||
+        HSE_UNLIKELY(
+            long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL))
+        return merr(EINVAL);
+
+    return 0;
+}
+
+
+int
+long_fn_call(char * socket_path, struct foo * foo)
+{
+    int n;
+
+    if (socket_path) {
+        n = strlcpy(
+            foo->foobar_foobar_foobar, some_long_function_name(socket_path),
+            sizeof(foo->foobar_foobar_foobar));
+        if (n >= sizeof(foo->foobar_foobar_foobar))
+            return -1;
+    }
+
+    return 0;
+}
+
+
+int
+labels(int x)
+{
+    if (x)
+        goto mpool_cleanup;
+
+    return 1;
+
+mpool_cleanup : {
+    int i;
+
+    for (i = 0; i < HSE_MCLASS_COUNT; i++) {
+        printf("%d", i);
+    }
+}
+
+    return 0;
+}
```

Comment:
colon should bind to the label

Reply:
I don't know why this is happening.  Seems like it got it right before.

Reply:
It's the open brace that confuses it.

Reply:
using `// eol` as showsn in `invalid2:` fixed the weirdness of `invalid3 : {`, but it created new weirdness.  i have no idea why.

Reply:
I would prefers labels being indented by 2, and live on a line by themselves...